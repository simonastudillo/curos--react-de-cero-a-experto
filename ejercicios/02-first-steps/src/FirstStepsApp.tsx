import { ItemCounter } from "./shopping-cart/ItemCounter";

export interface ItemInCart {
   productName: string;
   quantity: number;
}

const itemsInCart: ItemInCart[] = [
   {
      productName: 'Nintendo Switch 2', quantity: 1
   },
   {
      productName: 'Pro controller', quantity: 2
   },
   {
      productName: 'Super Smash Bros', quantity: 3
   },
]

export function FirstStepsApp() {
   return (
      <>
         <h1>Carrito de compras</h1>
         {
            itemsInCart.map(({ productName, quantity }) => (
               <ItemCounter key={productName} itemName={productName} quantity={quantity} />
            ))
         }
      </>
   );
}