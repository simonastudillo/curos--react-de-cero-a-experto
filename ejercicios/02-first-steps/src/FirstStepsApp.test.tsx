import { describe, expect, test } from "vitest";
import { FirstStepsApp } from "./FirstStepsApp";
import { render } from "@testing-library/react";


describe('FirstStepsApp', () => {

   test('should match snapshot', () => {
      // Arrange
      const { container } = render(<FirstStepsApp />);
      // Assert
      expect(container).toMatchSnapshot();
   });

});