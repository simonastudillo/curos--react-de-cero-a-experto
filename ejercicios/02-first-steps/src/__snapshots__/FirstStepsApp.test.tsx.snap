// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FirstStepsApp > should match snapshot 1`] = `
<div>
  <h1>
    Carrito de compras
  </h1>
  <section
    class="_itemRow_8b32a2"
  >
    <span
      class="_itemText_8b32a2"
      style="color: red;"
    >
      Nintendo Switch 2
    </span>
    <button>
      +1
    </button>
    <span>
      1
    </span>
    <button>
      -1
    </button>
  </section>
  <section
    class="_itemRow_8b32a2"
  >
    <span
      class="_itemText_8b32a2"
      style="color: black;"
    >
      Pro controller
    </span>
    <button>
      +1
    </button>
    <span>
      2
    </span>
    <button>
      -1
    </button>
  </section>
  <section
    class="_itemRow_8b32a2"
  >
    <span
      class="_itemText_8b32a2"
      style="color: black;"
    >
      Super Smash Bros
    </span>
    <button>
      +1
    </button>
    <span>
      3
    </span>
    <button>
      -1
    </button>
  </section>
</div>
`;
